### ISSUE TITLE
Monitor redis cache health

### ISSUE THREAD

**Issue by adberglund** [project-koku/koku#1807]
@adberglund to split into 2 issues where this issue will be the probes on the redis pod.

## User Story
As a developer I want to monitor when redis is unhealthy so that I can restart redis and application performance does not suffer.

## Impacts
- Operations

## Assumptions
- We can consider logging cache hits in the middleware for RBAC
- If we can monitor the cache performance directly, we should create or use an existing metric to alert on redis needing a restart
- Probably worth putting this ping check on our redis pod: https://gitlab.com/gitlab-org/kubernetes-gitlab-demo/blob/separate-prometheus/gitlab/redis-deployment.yml#L23-36
- Another option is to use the redis monitor in a thread in koku or somewhere. Then we could bump a redis error prometheus metric when we lose connection. See code snippet where a few calls to the redis-cli are logged, then I shut down the redis container to simulate the connection lose: 

```
>>> import redis
>>> r = redis.Redis()
>>> with r.monitor() as m:
...   for command in m.listen():
...     print(command)
...
{'time': 1582569867.263185, 'db': 0, 'client_address': '127.0.0.1', 'client_port': '41508', 'client_type': 'tcp', 'command': 'COMMAND'}
{'time': 1582569881.308555, 'db': 0, 'client_address': '127.0.0.1', 'client_port': '41508', 'client_type': 'tcp', 'command': 'keys *'}
{'time': 1582569891.686982, 'db': 0, 'client_address': '127.0.0.1', 'client_port': '41508', 'client_type': 'tcp', 'command': 'get worker'}
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.7/site-packages/redis/client.py", line 3321, in listen
    yield self.next_command()
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.7/site-packages/redis/client.py", line 3288, in next_command
    response = self.connection.read_response()
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.7/site-packages/redis/connection.py", line 734, in read_response
    response = self._parser.read_response()
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.7/site-packages/redis/connection.py", line 316, in read_response
    response = self._buffer.readline()
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.7/site-packages/redis/connection.py", line 248, in readline
    self._read_from_socket()
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.7/site-packages/redis/connection.py", line 193, in _read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
```

## Acceptance Criteria

- [ ] We know when redis is unhealthy and it either auto-restarts or we alert on it

**Comment by nbon12** [comment]
Sprint Planning Meeting Conversation:
Andrew: "A week ago we started to see slowdown in production, we restarted the Redis pod and everything sped back up. We don't have a way of detecting failures in Redis right now. We're using Django built-in cache framework. At a minimum we can consider logging cache hits in middleware for RBAC. In RBAC we can first check cache.  If we see some sort of error message we should trigger a count on a prometheus metric to get an alert. A liveness and readiness probe would be great for Redis, for example if Redis is down by the ping, then Redis gets restarted. Redis also has a monitor capability that can potentially log every command that gets sent to Redis. I don't know the root cause of our applications slowness, I want logging to help find out."
